evm/multisig.js

```diff
@@ -313,49 +314,43 @@ async function main(options) {
-        .makeOptionMandatory(true),
-);
-program.addOption(new Option('-p, --privateKey <privateKey>', 'private key').makeOptionMandatory(true).env('PRIVATE_KEY'));
-program.addOption(new Option('-y, --yes', 'skip deployment prompt confirmation').env('YES'));
-program.addOption(new Option('--offline', 'run script in offline mode'));
-program.addOption(new Option('--nonceOffset <nonceOffset>', 'The value to add in local nonce if it deviates from actual wallet nonce'));
-
-// options for setTokenMintLimits
-program.addOption(new Option('--symbols <symbols>', 'token symbols').makeOptionMandatory(false));
-program.addOption(new Option('--limits <limits>', 'token limits').makeOptionMandatory(false));
-
-// option for transferMintLimiter
-program.addOption(new Option('--mintLimiter <mintLimiter>', 'new mint limiter address').makeOptionMandatory(false));
-
-// options for withdraw
-program.addOption(new Option('--recipient <recipient>', 'withdraw recipient address').makeOptionMandatory(false));
-program.addOption(new Option('--withdrawAmount <withdrawAmount>', 'withdraw amount').makeOptionMandatory(false));
-
-// options for executeMultisigProposal
-program.addOption(new Option('--target <target>', 'execute multisig proposal target').makeOptionMandatory(false));
-program.addOption(new Option('--calldata <calldata>', 'execute multisig proposal calldata').makeOptionMandatory(false));
-program.addOption(new Option('--nativeValue <nativeValue>', 'execute multisig proposal nativeValue').makeOptionMandatory(false).default(0));
-
-program.action((options) => {
-    main(options);
-});
-
-program.parse();
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('multisig').description('Script to manage multisig actions');
+
+    addBaseOptions(program, { address: true });
+
+    program.addOption(new Option('-c, --contractName <contractName>', 'contract name').default('Multisig').makeOptionMandatory(false));
+    program.addOption(
+        new Option('--action <action>', 'multisig action')
+            .choices(['signers', 'setTokenMintLimits', 'transferMintLimiter', 'withdraw', 'executeMultisigProposal'])
+            .makeOptionMandatory(true),
+    );
+    program.addOption(new Option('--offline', 'run script in offline mode'));
+    program.addOption(new Option('--nonceOffset <nonceOffset>', 'The value to add in local nonce if it deviates from actual wallet nonce'));
+
+    // options for setTokenMintLimits
+    program.addOption(new Option('--symbols <symbols>', 'token symbols').makeOptionMandatory(false));
+    program.addOption(new Option('--limits <limits>', 'token limits').makeOptionMandatory(false));
+
+    // option for transferMintLimiter
+    program.addOption(new Option('--mintLimiter <mintLimiter>', 'new mint limiter address').makeOptionMandatory(false));
+
+    // options for withdraw
+    program.addOption(new Option('--recipient <recipient>', 'withdraw recipient address').makeOptionMandatory(false));
```

Comment:
take a look at `commander` package to see how you can do multi-level options.  For e.g. `[action] --flag [arg]`, i.e `action` becomes a positional arg instead of a flag. Not in scope for this PR, but might be cleaner to manage hierarchy of options to try this in the future.